test/metabase/test/data.clj

```diff
@@ -212,3 +213,48 @@
   `(-with-temp-db ~database-definition
      (fn [~db-binding]
        ~@body)))
+
+
+(defmacro expect-with-test-db
+  "Identical to `expect` but wraps the entire evaluation in `with-temp-db` so that both expected and actual statements
+   have access to the test data required for the test.
+
+     (expect-with-test-db [_ test-data]
+       (id :venues :id)
+       (sel :one :field [ForeignKey :destination_id] :origin_id (id :checkins :venue_id)))"
+  [test-db-binding expected actual]
+  (let [fn-name (gensym)]
+    `(def ~(vary-meta fn-name assoc :expectation true)
+       (fn []
+         (with-temp-db ~test-db-binding
+           (expectations/doexpect ~expected ~actual))))))
```

Comment:
You could simplify this macro a lot by just using `defn`:

```clojure
(defmacro expect-with-test-db [test-db-binding expected actual]
  `(defn fn# {:expectation true} []
     (with-temp-db ~test-db-binding
       (expectations/doexpect ~expected ~actual))))
```
*  `def + fn = defn`, you could just use that directly here
*  `defn` takes an optional metadata map param inbetween the (optional) docstr and the args. So no need to `vary-meta` 
*  Since you don't need to `vary-meta` you can just do an inline auto gensym

:+1: